modules/dnn/include/opencv2/dnn/dnn.hpp

```diff
@@ -100,6 +100,15 @@ CV__DNN_INLINE_NS_BEGIN
     CV_EXPORTS std::vector< std::pair<Backend, Target> > getAvailableBackends();
     CV_EXPORTS_W std::vector<Target> getAvailableTargets(dnn::Backend be);
 
+    /**
+     * @brief Enables detailed logging of the DNN model loading with CV DNN API.
+     * @param[in] isDiagnosticsMode Indicates whether diagnostic mode should be set.
+     *
+     * Diagnostic mode provides detailed logging of the model loading stage to explore
+     * potential problems (ex.: not implemented layer type).
```

Comment:
I propose to implicitly declare that the option is dangerous because it forces DNN module to skip internal shape/status/exceptions check and it may crash the app.

Reply:
Added 'note' about it.